server/src/main/java/org/elasticsearch/index/shard/ExplicitIndexSettingProvider.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.index.shard;
+
+import org.elasticsearch.common.settings.Settings;
+
+/**
+ * An {@link ExplicitIndexSettingProvider} is a provider for index level settings that can be set
+ * explicitly as a default value (so they show up as "set" for newly created indices)
+ */
+public interface ExplicitIndexSettingProvider {
```

Comment:
The naming here is a little confusing, because I thought this was about defaults, while "explicit" implies to me this is something else.

Reply:
It's short for "ExplicitDefaultIndexSettingProvider", and I should use the full name to be less confusing, I'll change this to use it.

Reply:
Maybe just IndexSettingsProvider? I'm still not understanding where the explicit comes in. I think the method could just be `getAdditionalIndexSettings`, similar to the `getAdditionalSettings()` we have on plugins for providing node settings.